### PR TITLE
rg-138: create table and save payment mathod

### PR DESCRIPTION
**Pull request by ValentinaTenus**
1. created payment method table
2. created subscription table
3. save data into these tables during first subscription
4. update subscription status with webhook events

### CODE REVIEW

**Review by Vitaliy-Kalenichenko — CHANGES_REQUESTED** [review]

File: backend/src/bundles/subscription/subscription.model.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import {
+    AbstractModel,
+    DatabaseTableName,
+} from '~/common/database/database.js';
+
+import { SubscriptionPlanModel } from '../stripe-events/models/subscription-plan.model.js';
+import { UserModel } from '../users/user.model.js';
+
+class SubscriptionModel extends AbstractModel {
+    public 'status': string;
```
Comment (jurok3x) [inline backend/src/bundles/subscription/subscription.model.ts:12]: `enum SubscriptionStatus {
    ACTIVE = 'active',
    CANCELED = 'canceled',
    TRIALING = 'trialing',
    PAST_DUE = 'past_due',
    INCOMPLETE = 'incomplete', 
    INCOMPLETE_EXPIRED = 'incomplete_expired',
    UNPAID = 'unpaid',
}`

perhaps enum could be better

File: backend/src/migrations/20240310061502_add_subscription_table.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+async function up(knex: Knex): Promise<void> {
+    await knex.raw('CREATE EXTENSION IF NOT EXISTS "uuid-ossp"');
+
+    await knex.schema.createTable(DatabaseTableName.SUBSCRIPTION, (table) => {
+        table
+            .uuid(DatabaseColumnName.ID)
+            .unique()
+            .primary()
+            .defaultTo(knex.raw(UUID_GENERATION_FUNCTION));
+        table.string(DatabaseColumnName.STATUS).notNullable();
```
Comment (jurok3x) [inline backend/src/migrations/20240310061502_add_subscription_table.ts:24]: `table.enum(DatabaseColumnName.STATUS, Object.values(SubscriptionStatus)).notNullable();`

**Review by vlad-blip — APPROVED** [review]

File: backend/src/bundles/subscription/subscription.model.ts (hunk 1 above)
Comment (ValentinaTenus) [reply backend/src/bundles/subscription/subscription.model.ts:12]: I can't use enum, because status'll be changed dynamically by stripe events.

Comment (jurok3x) [reply backend/src/bundles/subscription/subscription.model.ts:12]: Anyway, status has discrete values.

**Review by Valentyn13 — APPROVED** [review]

**Pull request closed without merging**